docs/data-preprocessor.rst

```diff
@@ -0,0 +1,75 @@
+Data Pre-Processor
+==================
+
+The preprocessor looks for the DIRECT_INCLUDE= keyword in the input yaml and
+concatenates the associated file at this point in the input file. The result
+is written to the output file or standard out if - is specified.
+
+It is expected that the keyword in the input yaml file will take the following
+format:
+
+.. code-block:: yaml
+
+   DIRECT_INCLUDE=/path/to/file/to/be/included
+
+Command line useage
+-------------------
+
+.. code-block:: bash
+
+   yp-preprocessor [options] -o output-file-name input-file-name
```

Comment:
Shouls this be:
` yp-preprocessor -i  input-file-name -o output-file-name [options]`

Reply:
I take your point on the order and have updated this as suggested (apart from the -i as per comment above)